addon/components/impagination-dataset.js

```diff
@@ -38,10 +38,14 @@ export default Ember.Component.extend({
 
     let readOffsetAttrFound = get(this, 'read-offset') >= 0;
     Ember.warn('Ember Impagination: `read-offset` attribute has been removed. Please use the `on-init` function instead.', !readOffsetAttrFound, {id: 'ember-impagination.attributes.read-offset'});
-
-    this.get('on-init')(this.get('model'));
+    this.fireOnInit();
```

Comment:
There is a subtle change in functionality. `fireOnInit()` is an observer and so it will fire the first time, but it will also fire every time the `dataset` changes. So if you have the following situation:

```hbs
{{#impagination-dataset fetch=fetch on-init=(action "setInitialReadOffset") as |dataset|}}
  {{!stuff}}
{{/impagination-dataset}}
```

If the `fetch` property changes, a completely new dataset gets created, and so you want to set the initial offset again to load the first pages within the load horizon.